charts/artifact-hub/templates/hub_deployment.yaml

```diff
@@ -64,8 +68,10 @@ spec:
             - name: http
               containerPort: 8000
               protocol: TCP
+          {{- with .Values.hub.deploy.resources }}
```

Comment:
At the moment the `resources` value for the hub is [marked as required in the schema](https://github.com/artifacthub/hub/blob/c73dafa519020415927665e14fb6eac1066120eb/charts/artifact-hub/values.schema.json#L312). With this change, it won't be anymore. I'm fine switching to `with` for resources, but if we go this way we should probably change the rest of the resources entries in other templates (tracker, scanner, etc) and not mark them as required anymore. Does that make sense?

Reply:
This is probably a general decision - in our clusters we require the declaration of resources but I know that there are many cases where you explicitly do not want to have resource definitions. Personally I am in favour of letting the user decide what she/he wants, so I would argue to make all occurrences optional.
But - this is my first time contribution to this project so I feel that this should be your decision :-)

Reply:
Let's make them all optional then 👍

Reply:
Done by my last commit :-)